docs/guides/schema-evolution.md

```diff
@@ -0,0 +1,196 @@
+
+You can evolve schema versions in `axual_topic_config` without changing the topic's schema, as long as the new schema version belongs to the same parent Schema.
+
+**Example: Adding an optional field**
+
+Original schema (v1.0.0):
+```json
+{
+  "type": "record",
+  "name": "GitOpsTestKey",
+  "namespace": "io.axual.qa.keys",
+  "fields": [
+    { "name": "keyId", "type": "string" }
+  ]
+}
+```
+
+Evolved schema (v2.0.0) - same `namespace + name`, new optional field:
+```json
+{
+  "type": "record",
+  "name": "GitOpsTestKey",
+  "namespace": "io.axual.qa.keys",
+  "fields": [
+    { "name": "keyId", "type": "string" },
+    { "name": "source", "type": ["null", "string"], "default": null }
+  ]
+}
+```
+
+Terraform configuration:
+```hcl
+# Original version
+resource "axual_schema_version" "key_v1" {
+  body    = file("schemas/key_v1.avsc")
+  version = "1.0.0"
+}
+
+# Evolved version - same parent Schema
+resource "axual_schema_version" "key_v2" {
+  body    = file("schemas/key_v2.avsc")
+  version = "2.0.0"
+}
+
+resource "axual_topic" "logs" {
+  key_schema = axual_schema_version.key_v1.schema_id  # Stays the same
+}
+
+resource "axual_topic_config" "logs_in_dev" {
+  topic              = axual_topic.logs.id
+  key_schema_version = axual_schema_version.key_v2.id  # Updated to v2
+}
+```
+
+This change requires only a single `terraform apply` because:
+- The topic's `key_schema` remains unchanged (same parent Schema)
+- Only the topic_config's `key_schema_version` is updated
+
+### Incompatible Schema Changes (Requires force=true)
+
+If you need to use a schema version that is not backward compatible (e.g., changed field types, removed fields, added required fields without defaults), you must set `force = true` on the `axual_topic_config` resource.
+
+**Example: Breaking changes**
+
+Incompatible schema - changed field type and added required field:
+```json
+{
+  "type": "record",
+  "name": "GitOpsTestKey",
+  "namespace": "io.axual.qa.keys",
+  "fields": [
+    { "name": "keyId", "type": "long" },
+    { "name": "requiredField", "type": "string" }
+  ]
+}
+```
+
+Terraform configuration:
+```hcl
+resource "axual_schema_version" "key_incompatible" {
+  body    = file("schemas/key_incompatible.avsc")
+  version = "99.0.0"
+}
+
+resource "axual_topic_config" "logs_in_dev" {
+  topic              = axual_topic.logs.id
+  key_schema_version = axual_schema_version.key_incompatible.id
+  force              = true  # Required for incompatible changes
+}
+```
+
+Without `force = true`, you will receive an error:
+```
+Incompatible schema_version change identified. Retry with force=true in the axual_topic_config resource
+```
+
+## Changing the Topic's Base Schema
+
+A topic once created with a specific key or value schema must continue to use the same schema for its lifetime. **If the schema itself needs changing then there is no option but to delete and recreate the topic.** This is by design and not a bug.
```

Comment:
Let's remove "This is by design and not a bug." since everything we mention here is implied to be by design.